frontend/src/scenes/funnels/funnelLogic.ts

```diff
@@ -24,22 +24,50 @@ import { FunnelStepReference } from 'scenes/insights/InsightTabs/FunnelTab/Funne
 import { eventDefinitionsModel } from '~/models/eventDefinitionsModel'
 import { personsModalLogic } from 'scenes/trends/personsModalLogic'
 
+function aggregateBreakdownResult({
+    result: breakdownList,
+    ...apiResponse
+}: FunnelResult<FunnelStep[][]>): FunnelResult<MappedFunnelStep[]> {
+    let result: MappedFunnelStep[] = []
+    if (breakdownList.length) {
+        result = breakdownList[0].map((step, i) => ({
+            ...step,
+            breakdown_value: step.breakdown,
+            count: breakdownList.reduce((total, breakdownSteps) => total + breakdownSteps[i].count, 0),
+            breakdown: breakdownList.reduce((allEntries, breakdownSteps) => [...allEntries, breakdownSteps[i]], []),
+            average_conversion_time: null, // TODO API needs to return a weighted average
+            people: [], // We could concatenate and de-dupe the UUIDs from breakdown values here, but it would be more performant to do it on backend
+        }))
+    }
+    return {
+        ...apiResponse,
+        result,
+    }
+}
+
 function wait(ms = 1000): Promise<any> {
     return new Promise((resolve) => {
         setTimeout(resolve, ms)
     })
 }
 
 const SECONDS_TO_POLL = 3 * 60
-
+interface FunnelRequestParams extends FilterType {
+    refresh?: boolean
+    from_dashboard?: boolean
+    funnel_window_days?: number
+}
 interface TimeStepOption {
     label: string
     value: number
     average_conversion_time: number
     count: number
 }
 
-async function pollFunnel(params: Record<string, any>): Promise<FunnelResult & FunnelsTimeConversionResult> {
+async function pollFunnel(
+    params: FunnelRequestParams
+): Promise<FunnelResult & FunnelResult<FunnelStep[][]> & FunnelResult<number[]>> {
+    // Tricky: This API endpoint has wildly different return types depending on parameters.
```

Comment:
Could you clarify? Or better yet, document with types? Function overloading is possible with TypeScript.